### DISCUSSION TITLE
Heap Space Issue while using SFTP as a Source

### DISCUSSION THREAD

**Discussion by rjkroll** [nextgenhealthcare/connect#5959] [Q&A]
We have a channel that is just a simple SFTP source that goes out and downloads whatever CSV file is at the endpoint.  

It had been working fine for years, but recently some of the files started growing rather large (hundreds of MB).    We had the Administrator, and server options, set to 2GB and on the channel, turned of Logging at all to try and cut down on memory use.

The last time it polled, there was 3 files.   It downloaded 2 without issues, then had a Java Heap space issue on the last file.    When the Heap space issue happened, the file stopped downloading and 'disappeared'.   It isn't in any folder, trash bin, any where on the server.

My question is, when this happens, is the file just lost?  Or does it get somehow moved to somewhere?    I am already prepared to have lost this file and asked them to resend (also moved away from using Mirth to download large files), but can does anyone know/can confirm that in these instances, the file is truly lost?

**Comment by jonbartels** [comment]
Show two things:
1. Show your message storage settings - https://docs.nextgen.com/bundle/Mirth_User_Guide_40/page/connect/connect/topics/c_Message_Storage_Settings_connect_ug.html
2. Show your file conntector settings for error and move to
3. Search `mirth.log` for the channel ID anything there?
4. Search the log for "message recovery". Find anything?

I'm not able to articulate the details very well, but your message should not be lost EXCEPT you got an out of memory error. So if you got an OOM when the message was processing, you also likely got an OOM when the message recovery kicked in. OOMs in message recovery are not fatal but if recovery fails Mirth won't try again.

It may be possible to find the message in the Mirth DB. What DB engine are you using?
